sdk/protocol-manager-poc/src/interfaces/protocolPlugin.ts

```diff
@@ -276,22 +265,47 @@ export const createMakerPlugin: CreateProtocolPlugin<MakerPoolId> = (ctx: Protoc
         },
         getPosition: async (positionId: IPositionId): Promise<Position> => {
             throw new Error("Not implemented")
+        },
+        _validate: function (candidate: unknown): asserts candidate is MakerPoolId {
+            if (typeof candidate !== 'object' || candidate === null) {
+                throw new Error("Candidate is not an object");
+            }
+
+            if ('protocol' in candidate && 'ilkType' in candidate) {
+                const { protocol, ilkType } = candidate as { protocol: unknown, ilkType: unknown };
+
+                if (typeof protocol !== 'object' || protocol === null) {
+                    throw new Error("Protocol is not an object");
+                }
+
+                if ('name' in protocol && protocol.name !== ProtocolName.Maker) {
+                    throw new Error("Protocol is not Maker");
+                }
+
+                if (typeof ilkType !== 'string' || !assertStringIsIlkType(ilkType)) {
+                    throw new Error("IlkType is invalid");
+                }
+            } else {
+                throw new Error("Candidate does not have the required properties");
+            }
         }
     }
 
     return plugin
 }
 
+
+function assertStringIsIlkType(maybeIlk: string): boolean {
```

Comment:
as for naming used for this function, this seem like a simple condition check, from the assertion I would expect behavior to halt the execution by throwing en error
just a comment to align on naming things

Reply:
No issue from me. I've updated both to `is<Thing>` checks